crates/sui-core/src/consensus_adapter.rs

```diff
@@ -1065,11 +1061,10 @@ impl ConsensusAdapter {
                 .await
             {
                 Err(err) => {
-                    // This can happen during reconfig, or when consensus has full internal buffers
-                    // and needs to back pressure, so retry a few times before logging warnings.
-                    if retries > 30 || (retries > 3 && (is_soft_bundle || !is_dkg)) {
+                    // This can happen during reconfig, so keep retrying until succeed.
+                    if cfg!(msim) || retries > 3 {
                         warn!(
-                            "Failed to submit transactions {transaction_keys:?} to consensus: {err:?}. Retry #{retries}"
+                            "Failed to submit transactions {transaction_keys:?} to consensus: {err}. Retry #{retries}"
```

Comment:
In the tests did you observe this being ok during epoch changes? Just want to make sure it won't be a "noisy" log